plugins/optimization-detective/detect.js

```diff
@@ -127,6 +127,12 @@ function recursiveFreeze( obj ) {
 	Object.freeze( obj );
 }
 
+// This needs to be captured early in case the user later resizes the window.
+const viewport = {
+	width: win.innerWidth,
+	height: win.innerHeight,
+};
+
```

Comment:
I added this here to capture the initial width/height when the page loads. Later when about to submit the metrics for storage, it checks to see if the viewport width or height has changed, and then it aborts. This is important because if the viewport window size changes during the life of the page, all bets are off as to what is the expected LCP element.

Reply:
This seems to be a separate fix, that relates to the problem this PR is solving, but is relevant on its own too? Ideally this kind of thing would be its own PR (and then you could use sub-PRs), as it's really difficult to review large PRs like this with multiple changes :)

Reply:
You're right, it was a separate fix. But it was something that tripped me up when I was working on this PR. I've just cherry-picked it into a new PR: https://github.com/WordPress/performance/pull/1712

Once you review and merge that, I'll merge `trunk` back into this branch to remove it from the diff.

Reply:
Now eliminated from the diff.